account_payment_cancel/tests/test_payment_cancel.py

```diff
@@ -0,0 +1,145 @@
+        self.order_name = '2017/1013'
+        self.journal_name = '2017/1013'
+        self.payment_line_name = 'Ltest'
+
+        # Create payment order from the invoice
+        invoice = self.env['account.invoice'].search(
+            [('move_name', '=', self.invoice_name)])
+
+        # Validate the invoice
+        invoice.action_invoice_open()
+
+        # Create a payment order
+        action = invoice.create_account_payment_line()
+        payment_order_id = action['res_id']
+
+        payment_order = self.env['account.payment.order'].search(
+            [('id', '=', payment_order_id)])
+
+        partner_bank = self.env['account.journal'].search(
+            [('name', '=', self.journal_name)])
+
+        bank = self.env['account.journal'].search([('name', '=', 'Bank')])
+
+        payment_order.name = self.order_name
+        bank.update_posted = True
+
+        payment_order.journal_id = partner_bank.id
+        # Confirm payment order
+        payment_order.draft2open()
+        # Generate payment file
+        payment_order.open2generated()
+        # File successfully uploaded
+        payment_order.generated2uploaded()
+
+        payment_line = self.env['bank.payment.line'].search(
+            [('order_id', '=', payment_order.id)])
+
+        payment_line.name = self.payment_line_name
+
+
+    #
+        # self.env.cr.commit()
+    #
+    def test_invoice_exist(self):
+        invoice = self.env['account.invoice'].search(
+            [('move_name', '=', self.invoice_name)])
+
+        self.assertTrue(invoice)
+
+    def test_invoice_line_exist(self):
+        invoice_line = self.env['account.invoice.line'].search(
+            [('name', '=', self.invoice_line_name)])
+
+        self.assertTrue(invoice_line)
+
+    def test_invoice_state_is_paid(self):
+        invoice = self.env['account.invoice'].search(
+            [('move_name', '=', self.invoice_name)])
+
+        self.assertEqual(invoice.state, 'paid')
+
+    def test_payment_order_exist(self):
+        payment_order = self.env['account.payment.order'].search(
+            [('name', '=', self.order_name)])
+
+        self.assertTrue(payment_order)
+
+    def test_payment_order_state_is_uploaded(self):
+        payment_order = self.env['account.payment.order'].search(
+            [('name', '=', self.order_name)])
+
+        self.assertEqual(payment_order.state, 'uploaded')
+
+    def test_account_move_exist(self):
+        payment_order = self.env['account.payment.order'].search(
+            [('name', '=', self.order_name)])
+        account_move = self.env['account.move'].search(
+            [('payment_order_id', '=', payment_order.id)])
+
+        self.assertTrue(account_move)
+
+    def test_payment_line_exist(self):
+        payment_line = self.env['bank.payment.line'].search(
+            [('name', '=', self.payment_line_name)])
+
+        # Check if there is a payment line
+        self.assertTrue(payment_line)
```

Comment:
You can remove all the tests up to here. There is no point testing the data you created.